src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -0,0 +1,332 @@
+    private static final String MESSAGE_ERROR_INVALID_ITEM_NUM = "Please input the item number as a number! E.g 1";
+    private static final String MESSAGE_ERROR_NOT_A_NUMBER = "Please input a number!";
+
+
+    /**
+     * Returns the correct command to be executed depending on user input.
+     * Command words are case-insensitive.
+     *
+     * @param input Raw user input string
+     * @return Command class representing the correct command to be executed
+     */
+    public Command parseCommand(String input) {
+        final String[] commandAndParams = splitInputIntoCommandAndParams(input);
+        final String commandWord = commandAndParams[0].toLowerCase(); //case-insensitive (all lower case)
+        final String params = commandAndParams[1];
+
+        switch (commandWord) {
+        case Command.COMMAND_WORD_ADD:
+            return parseAddItems(params);
+        case Command.COMMAND_WORD_DELETE:
+            return parseDeleteItems(params);
+        case Command.COMMAND_WORD_VIEW:
+            return parseViewItems(params);
+        case Command.COMMAND_WORD_BMI:
+            return parseBmi(params);
+        case ChangeNameCommand.COMMAND_WORD:
+            return new ChangeNameCommand(params);
+        case ChangeHeightCommand.COMMAND_WORD:
+            return parseChangeHeight(params);
+        case ChangeWeightCommand.COMMAND_WORD:
+            return parseChangeWeight(params);
+        case CreateProfileCommand.COMMAND_WORD:
+            return parseCreateProfile(params);
+        case SetGoalCommand.COMMAND_WORD:
+            return parseSetGoal(params);
+        case OverviewCommand.COMMAND_WORD:
+            return new OverviewCommand();
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case HelpCommand.COMMAND_WORD:
+            return new HelpCommand();
+        default:
+            return new InvalidCommand(MESSAGE_ERROR_COMMAND_DOES_NOT_EXIST);
+        }
+
+    }
+
+    private Command parseAddItems(String params) {
+        try {
+            final String itemTypePrefix = extractItemTypePrefix(params);
+            final String description = extractItemDescription(params, itemTypePrefix);
+            final int calories = extractItemCalories(params);
+
+            if (itemTypePrefix.equals(Command.COMMAND_PREFIX_EXERCISE)) {
+                return new AddExerciseCommand(description, calories);
+            } else {
+                return new AddFoodCommand(description, calories);
+            }
+        } catch (ItemNotSpecifiedException e) {
+            return new InvalidCommand(String.format(Command.MESSAGE_ERROR_ITEM_NOT_SPECIFIED,
+                    AddFoodCommand.MESSAGE_COMMAND_FORMAT,
+                    AddExerciseCommand.MESSAGE_COMMAND_FORMAT));
+        } catch (ParamInvalidException e) {
+            return new InvalidCommand(e.getMessage());
+        }
+    }
+
+    private Command parseDeleteItems(String params) {
```

Comment:
I was thinking that we can allow user to delete all foods and exercises using just one command, maybe call `delete f/all` or `delete e/all` in the future, so that user no need to delete one by one, so perhaps you can consider allow user to pass `all` string, we can discuss this further in the meeting :)

Reply:
i think that this is a not bad suggestion, i will edit ExerciseList class to clear array list if we are proceeding with this idea

Reply:
Good idea, could add this as another user story! 👍